### PR TITLE
feat: added management command, set is_marketable to False

### PR DESCRIPTION
**Pull request by DmytroAlipov**
### Description:
Added management command to set is_marketable to False for CourseRunType instances that have this parameter set to True.
This parameter is [set to True by default](https://github.com/openedx/course-discovery/blob/ea6e542f270a78ce2b26cfc51bccaec27a1853ab/course_discovery/apps/course_metadata/models.py#L539). This parameter [affects the formation of marketing_url](https://github.com/openedx/course-discovery/blob/ea6e542f270a78ce2b26cfc51bccaec27a1853ab/course_discovery/apps/course_metadata/models.py#L1949) for the CourseRun instance. These links are not always needed as not everyone in the community uses marketing sites for courses. In this case, the presence of this link leads to the creation of invalid links. For example in the cart (example for nutmeg release, but the same behavior for later versions.): 
Screen_34

Screen_35

Screen_36

If is_marketable=False, then this link is a [link to the about page](https://github.com/openedx/ecommerce/blob/a4feac1a2b6da575064898bbade4644fe79dd7aa/ecommerce/extensions/basket/views.py#L355) it is a course. The presence of this link is the solution to this problem.

**It is corrected manually through the discovery admin panel:**
Screen_30

**However, the automation of this process allows you not to forget about this moment with each deployment.**

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @DmytroAlipov!  Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket as you can:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


Please let us know once your PR is ready for our review and all tests are green.

**Comment by e0d** [comment]
@DmytroAlipov I've updated this PR from master with a merge commit.  Tests running.

**Comment by DmytroAlipov** [comment]
@e0d Hi! I fixed the import order in my test. 
But the failed `course_discovery/apps/course_metadata/data_loaders/tests/test_api.py::ProgramsApiDataLoaderTests::test_ingest` test fails even on the master without my changes.
The issue with **assert_api_called**.

**Comment by e0d** [comment]
Updating from master and running the tests.

**Comment by e0d** [comment]
@DmytroAlipov strange, [this recent PR](https://github.com/openedx/course-discovery/pull/3834) doesn't seem to be having the issue, can you make sure you've pulled upstream and try a local rebase to see if that resolves the issue?

**Comment by DmytroAlipov** [comment]
@e0d Hi! Indeed, after updating the master branch and rebase, all tests pass on the local instance. Please run the pipelines again. I hope this time everything goes well.

**Comment by e0d** [comment]
Tests approved.

**Comment by DmytroAlipov** [comment]
@e0d I fixed the problem with the wrong sorting of imports.

**Comment by e0d** [comment]
Tests approved

**Comment by DmytroAlipov** [comment]
@e0d I don't understand why it's not passing the check... I sorted the import.

**Comment by e0d** [comment]
Approved, will take a close look if there are further failures.  Is  `make quality` working locally?

**Comment by e0d** [comment]
@DmytroAlipov my local quality isn't running correctly, but I think the change you need is:

```
diff --git a/course_discovery/apps/course_metadata/management/commands/change_is_marketable_to_false.py b/course_discovery/apps/course_metadata/management/commands/change_is_marketable_to_false.py
index 71ae14d6f..3a769059d 100644
--- a/course_discovery/apps/course_metadata/management/commands/change_is_marketable_to_false.py
+++ b/course_discovery/apps/course_metadata/management/commands/change_is_marketable_to_false.py
@@ -1,7 +1,7 @@
 import logging
 
-from course_discovery.apps.course_metadata.models import CourseRunType
 from django.core.management import BaseCommand
+from course_discovery.apps.course_metadata.models import CourseRunType
 
 logger = logging.getLogger(__name__)
 
diff --git a/course_discovery/apps/course_metadata/management/commands/tests/test_change_is_marketable_to_false.py b/course_discovery/apps/course_metadata/management/commands/tests/test_change_is_marketable_to_false.py
index 1a0d11f9a..86d99db5b 100644
--- a/course_discovery/apps/course_metadata/management/commands/tests/test_change_is_marketable_to_false.py
+++ b/course_discovery/apps/course_metadata/management/commands/tests/test_change_is_marketable_to_false.py
@@ -1,8 +1,9 @@
-from course_discovery.apps.course_metadata.models import CourseRunType
-from course_discovery.apps.course_metadata.tests.factories import CourseRunTypeFactory
 from django.core.management import call_command
 from django.test import TestCase
 
+from course_discovery.apps.course_metadata.models import CourseRunType
+from course_discovery.apps.course_metadata.tests.factories import CourseRunTypeFactory
+
 
 class ChangeIsMarketableToCourseRunTypesCommandTests(TestCase):
     def setUp(self):

```

**Comment by DmytroAlipov** [comment]
@e0d, I changed the import

**Comment by DmytroAlipov** [comment]
@e0d I don't understand anything. I ran the test `course_discovery/apps/api/v1/tests/test_views/test_currency.py::TestCurrencyCurrencyView::test_get - assert 2 == 1` - the test passed successfully. I updated my local branch, ran the same test again, and it passed successfully again. I don't understand why this test is not passing in the repository.

**Comment by e0d** [comment]
@DmytroAlipov looks like we're all green.

**Comment by DmytroAlipov** [comment]
@e0d excellent news! I have already started to go crazy with these checks)

**Comment by mphilbrick211** [comment]
Hi @kashifch! Would you mind taking a look at this and merging if it looks good to you?

In the meantime, @DmytroAlipov, would you mind fixing the branch date issue? Thanks!

**Comment by DmytroAlipov** [comment]
@mphilbrick211 Done ;)

**Comment by e0d** [comment]
Updating with a merge commit and running the tests.

**Comment by mphilbrick211** [comment]
Hi @kashifch! Would you mind taking a look at this and merging if all looks good? Thanks!

**Comment by mphilbrick211** [comment]
> Hi @kashifch! Would you mind taking a look at this and merging if all looks good? Thanks!

Friendly ping on this @kashifch :)

**Comment by mphilbrick211** [comment]
Hi @DmytroAlipov - it looks like a failing test has popped up. Would you mind taking a look?

**Comment by e0d** [comment]
@DmytroAlipov if you have a moment, could you push a trivial commit to this branch?  I'd like to test something related to the Github actions and outside committers.

**Comment by DmytroAlipov** [comment]
@e0d done ;)

**Comment by DmytroAlipov** [comment]
@mphilbrick211 Hi. It's just a friendly ping, all tests passed ;)

**Comment by mphilbrick211** [comment]
> Author

Thanks @DmytroAlipov :)

@AliAdnanSohail would you please review and merge if all looks good to you? If you could please also handle the branch date item when merging, that would be great. Thanks!

**Review by AliAdnanSohail — APPROVED** [review]

**Review by Ali-D-Akbar — APPROVED** [review]
Approving but I don't see very much of a usage for this command since we have a very limited number of course run types anyways and is_marketable can be changed through admin for these.

**Comment by DawoudSheraz** [comment]
> Approving but I don't see very much of a usage for this command since we have a very limited number of course run types anyways and is_marketable can be changed through admin for these.

Totally agree with this. We can use [admin actions](https://docs.djangoproject.com/en/4.2/ref/contrib/admin/actions/) instead of the management command. It will make it easier to manage this from admin & would not require going to server to run this command.

**Comment by DmytroAlipov** [comment]
Hi @DawoudSheraz, @Ali-D-Akbar 
- Limited Course Run Types: I understand that we have a limited number of course run types, and it is possible to update the is_marketable attribute through the admin interface. However, I believe that implementing a management command would be beneficial for automation purposes. By having a dedicated command, we can ensure a streamlined and efficient process for updating the attribute, even with a limited number of course run types. It allows us to automate the task and eliminates the need for manual intervention, ultimately saving time and effort.

- Simplified Administration: I agree that using admin actions can be a viable option for modifying the is_marketable attribute directly from the admin interface. However, I think it would be more advantageous to develop a management command. With a command in place, administrators can still make changes through the admin interface if needed. However, having a dedicated command provides an additional layer of automation and simplifies the overall administration process. It allows us to handle the attribute update centrally, without requiring individuals to access the server and run a command separately. This approach aligns with our goal of automating tasks and ensures a consistent and efficient maintenance policy.

In summary, by developing a management command, we can achieve automation, simplify the deployment process, and adhere to our maintenance policies effectively.

**Comment by mphilbrick211** [comment]
Hi @AliAdnanSohail! Are you able to review/merge this for us? Thanks!

**Comment by openedx-webhooks** [comment]
@DmytroAlipov 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/course-discovery&entry.867055334=https://github.com/openedx/course-discovery/pull/3810&entry.1484655318=https://github.com/DmytroAlipov&entry.752974735=2023-02-22+12:11&entry.1917517419=2023-06-22+14:44&entry.2133058324=Yes) so we can improve your experience in the future.
